### PR TITLE
CAV-191 Adding metrics to validation service

### PR DESCRIPTION
**Pull request by mar36tin**

### CODE REVIEW

File: app/uk/gov/hmrc/cipphonenumbervalidation/service/ValidateService.scala [hunk 1]
```diff
@@ -42,12 +44,19 @@ class ValidateService @Inject()(phoneNumberUtil: PhoneNumberUtil) extends Loggin
         case _ if phoneNumber.isEmpty || existsLetter(phoneNumber) || containsChars(phoneNumber) ||
           !mandatoryFirstChars.contains(phoneNumber.charAt(0)) => false
         case _ if isValidPhoneNumber(phoneNumber) =>
+
+          val phoneNumberType = getPhoneNumberType(phoneNumber).name
+          metricsService.recordMetric(phoneNumberType)
```
Comment (brdgeth) [inline app/uk/gov/hmrc/cipphonenumbervalidation/service/ValidateService.scala:49]: can we verify the correct phone number type is sent to the recordMetrics please

File: app/uk/gov/hmrc/cipphonenumbervalidation/service/ValidateService.scala [hunk 2]
```diff
@@ -42,12 +44,19 @@ class ValidateService @Inject()(phoneNumberUtil: PhoneNumberUtil) extends Loggin
...
+            phoneNumberType.toLowerCase.capitalize)
       }
     } match {
       case Success(phoneNumberResponse: PhoneNumberResponse) => Future.successful(Ok(Json.toJson(phoneNumberResponse)))
-      case Success(false) | Failure(_) =>
+      case Success(false) =>
+        logger.warn("Failed to validate phone number")
+        Future.successful(BadRequest(Json.toJson(ErrorResponse("VALIDATION_ERROR", "Enter a valid telephone number"))))
+      case Failure(e) =>
+        metricsService.recordMetric(e.toString)
```
Comment (brdgeth) [inline app/uk/gov/hmrc/cipphonenumbervalidation/service/ValidateService.scala:59]: can we verify the correct failure is sent to the recordMetrics please

**Review by brdgeth — COMMENTED** [review]
minor fine tuning of the unit testing

**Comment by johno1985** [comment]
rebuild

**Comment by johno1985** [comment]
outstanding merge conflicts

**Pull request closed without merging**